snippets/csharp/System.Collections.Generic/ICollectionT/Overview/program.cs

```diff
@@ -198,7 +198,7 @@ public void CopyTo(Box[] array, int arrayIndex)
            throw new ArgumentNullException("The array cannot be null.");
         if (arrayIndex < 0)
            throw new ArgumentOutOfRangeException("The starting array index cannot be negative.");
-        if (Count > array.Length - arrayIndex + 1)
+        if (Count > array.Length - arrayIndex)
```

Comment:
please add a test case and make sure it fails before this change and passes after